### PR TITLE
Add NullOrUndefined handling

### PR DESCRIPTION
**Pull request by jacereda**

### CODE REVIEW

**Comment by justinwoo** [comment]
I don't know if we should really do this since it destroys the round trip from stringify though, due to the Maybe type inside of NullableOrUndefined...

**Comment by justinwoo** [comment]
But yeah, maybe still worth taking, as long as stringifying it back out isn't at all a concern. Thoughts?

File: test/Main.purs [hunk 1]
```diff
@@ -37,3 +39,8 @@ main = run [consoleReporter] do
         { "c": 1, "d": 2}
       """
       isRight (result :: E MyTest) `shouldEqual` false
+    it "works with JSON containing NullOrUndefined" do
+      let result = handleJSON """
+        { "a": 1, "b": "asdf", "c": true, "d": ["A", "B"], "e": ["C", "D"]}
+      """
+      isRight (result :: E MyTest) `shouldEqual` true
```
Comment (justinwoo) [inline test/Main.purs]: Btw, might be nicer to split this type since it effects the first test too

**Comment by jacereda** [comment]
Do you  refer to this `stringify`?

https://pursuit.purescript.org/packages/purescript-ffi-utils/2.0.0/docs/FFI.Util#v:stringify

**Comment by justinwoo** [comment]
No, something like [this](http://try.purescript.org/?gist=2720825b0476c8924717437fb6f6eefb&session=cb61df9f-9070-9ea2-bbe7-06b6e465050d) that would use RowToList to figure out if the fields are convertable to to foreign

**Comment by jacereda** [comment]
Perhaps a note warning about the problem would suffice. In my case I'm only interested in reading so far.

In any case, couldn't `NullOrUndefined` fields be handled when serialising?

**Comment by justinwoo** [comment]
Sounds like some kind of uncanny valley where people should really be using Foreign-Generics or something instead, but let's see what happens.

**Pull request merged**